### ISSUE TITLE
dapp build --extract puts new line at end of .bin files

### ISSUE THREAD

**Issue by ethernomad** [dapphub/dapptools#246]
Previously `dapp build` would create .bin files that were not terminated with a new line.

Now `dapp build --extract` puts a new line at the end of every .bin file. This can cause problems with deployment.

**Comment by rainbreak** [comment]
Which tooling does this cause a problem with? Is the newline being interpreted as hex `0x0a` and appended to the code?

**Comment by ethernomad** [comment]
If I read the file and try to deploy it with web3.js it complains that it is not valid hex.
I just added a .trim() to fix it.

https://github.com/mix-blockchain/mix-acuity/blob/b95a9325239512bfa80745045a327865ac9b9e7e/src/lib/MixAccount.js#L71

**Comment by rainbreak** [comment]
@gbalabasquer have you noticed this as a problem at all in deploying things?

**Comment by gbalabasquer** [comment]
@rainbreak not really. We were able to successfully deploy contracts using `dapp create` when generating files with `dapp build --extract`. But haven't used the files for anything else.

**Comment by asymmetric** [comment]
If we leave the newline in I think we should document it in the CHANGELOG.
